vnext/local-cli/runWindows/utils/autolink.js

```diff
@@ -6,95 +6,341 @@
-      ? windowsPlatformConfig.cppProjFile
-      : null;
-    if (cppProjFile == null) {
-      console.log(
-        'AutoLink currently is only supported on C++/WinRT main project.',
-      );
-      return;
+    verboseMessage(output, verbose);
+
+    verboseMessage('Parsing output...', verbose);
+
+    const rnConfig = JSON.parse(output);
+    const windowsAppConfig = rnConfig.project.windows;
+
+    if (!windowsAppConfig) {
+      throw 'Windows auto-link only supported on windows app projects.';
     }
-    console.log(`cppProjFile: ${cppProjFile}`);
 
-    const sourceDir = config.project.windows.sourceDir;
-    console.log(`sourceDir: ${sourceDir}`);
+    verboseMessage('Found Windows app project, parsing...', verbose);
 
-    const projectName = config.project.windows.projectName;
-    console.log(`projectName: ${projectName}`);
+    const solutionFile = path.join(
+      windowsAppConfig.folder,
+      windowsAppConfig.sourceDir,
+      windowsAppConfig.solutionFile,
+    );
 
-    //#1. update nativeModules.g.h
-    const generatedHeader = path.join(
-      sourceDir,
-      projectName,
-      'AutolinkedNativeModules.g.h',
+    const projectFile = path.join(
+      windowsAppConfig.folder,
+      windowsAppConfig.sourceDir,
+      windowsAppConfig.project.projectFile,
     );
-    if (!fs.existsSync(generatedHeader)) {
-      console.log(
-        'AutoLink can not locate generated header file: {generatedHeader}',
+
+    const projectDir = path.dirname(projectFile);
+    const projectLang = windowsAppConfig.project.projectLang;
+
+    verboseMessage('Parsing dependencies...', verbose);
+
+    const dependencies = rnConfig.dependencies;
+
+    let windowsDependencies = {};
+
+    for (const dependencyName in dependencies) {
+      const windowsDependency = dependencies[dependencyName].platforms.windows;
+
+      if (windowsDependency) {
+        verboseMessage(
+          `Found dependency ${chalk.bold(dependencyName)}.`,
+          verbose,
+        );
+        windowsDependencies[dependencyName] = windowsDependency;
+      }
+    }
+
+    // Generating cs/h files for app code consumption
+    if (projectLang === 'cs') {
+      let csUsingNamespaces = '';
+      let csReactPacakgeProviders = '';
+
+      for (const dependencyName in windowsDependencies) {
+        windowsDependencies[dependencyName].projects.forEach(project => {
+          if (project.directDependency) {
+            csUsingNamespaces += `\n\n// Namespaces from ${dependencyName}`;
+            project.csNamespaces.forEach(namespace => {
+              csUsingNamespaces += `\nusing ${namespace};`;
+            });
+
+            csReactPacakgeProviders += `\n            // IReactPackageProviders from ${dependencyName}`;
+            project.csPackageProviders.forEach(packageProvider => {
+              csReactPacakgeProviders += `\n            packageProviders.Add(new ${packageProvider}());`;
+            });
+          }
+        });
+      }
```

Comment:
i may have said this elsewhere but please consider making this a templated file like how we do in the app template code :)